### PR TITLE
Fixed build issues in v7

### PR DESCRIPTION
**Pull request by sriteja777**
## Proposed changes
The build in v7 is [failing](https://github.com/webdriverio/webdriverio/issues/10084) due to dependency upgrades for [typescript](https://github.com/webdriverio/webdriverio/pull/10009) and [@types/archiver](https://github.com/webdriverio/webdriverio/pull/9948)

For `@types/archiver`
The version 5.3.2 has removed the module glob, so had to install the `@types/glob` package manually where `glob` is used.

For `typescript`
As it is major version upgrade, it introduced some new errors, have fixed them.

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [ ] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

**Comment by christian-bromann** [comment]
Thanks for providing a PR with build fixes. There still seem to be some linting issues 🤔

File: packages/wdio-cli/src/constants.ts [hunk 1]
```diff
@@ -186,13 +186,13 @@ export const QUESTIONNAIRE = [{
     type: 'input',
     name: 'hostname',
     message: 'What is the host address of that cloud service?',
-    when: /* istanbul ignore next */ (answers: Questionnair) => answers.backend.toString().indexOf('different service') > -1
+    when: /* istanbul ignore next */ (answers: Questionnair) => String(answers.backend).indexOf('different service') > -1
```
Comment (christian-bromann) [inline packages/wdio-cli/src/constants.ts:189]: Can we do this instead everywhere? This seems to be a type issue and if we just case the type the compiled code stays the same.

```suggestion
    when: /* istanbul ignore next */ (answers: Questionnair) => (answers.backend as string).indexOf('different service') > -1
```

File: packages/wdio-cli/src/utils.ts [hunk 2]
```diff
@@ -443,7 +443,7 @@ export function getPathForFileGeneration (answers: Questionnair) {
 }
 
 export function getDefaultFiles (answers: Partial<Questionnair>, filePath: string) {
-    return answers?.isUsingCompiler?.toString().includes('TypeScript')
+    return String(answers?.isUsingCompiler).includes('TypeScript')
```
Comment (christian-bromann) [inline packages/wdio-cli/src/utils.ts:446]:
```suggestion
    return (answers?.isUsingCompiler as string).includes('TypeScript')
```

File: packages/wdio-cli/src/constants.ts (hunk 1 above)
Comment (sriteja777) [reply packages/wdio-cli/src/constants.ts]: Sure done!

File: packages/wdio-cli/src/utils.ts (hunk 2 above)
Comment (sriteja777) [reply packages/wdio-cli/src/utils.ts]: done!

**Comment by sriteja777** [comment]
@christian-bromann Have fixed the linting issue. But currently there is some other error in test. Any idea of this error? 
    
    Error: ../../../packages/wdio-mocha-framework/node_modules/expect-webdriverio/types/standalone.d.ts(4,73): error 
    TS2344: Type 'R' does not satisfy the constraint 'void | Promise<void>'.
    Type 'R' is not assignable to type 'Promise<void>'.

**Comment by christian-bromann** [comment]
Such a bug has been reported already: https://github.com/webdriverio/expect-webdriverio/pull/962

**Comment by sriteja777** [comment]
@christian-bromann as that is a bug in running unit tests, but the build is working fine. Can we merge this PR if it looks good to you?

**Comment by sriteja777** [comment]
The build works fine (previously it used to failing) when run locally on mac 😅  and also the unit tests passes fine (ran `npm run-s test:coverage`).

Only the `test:typings` fail with the above mentioned issue. I am not exactly sure though on how to confirm that it is related to https://github.com/webdriverio/expect-webdriverio/pull/962.

**Comment by christian-bromann** [comment]
I think this error was introduced with the update to TypeScript v5. If we can resolve it within here or by making changes to the `expect-webdriverio` v3 branch (which is the version we use for WebdriverIO v7 and below).

**Comment by sriteja777** [comment]
@christian-bromann Have made some typings changes to `expect-webdriverio` in this [PR](https://github.com/webdriverio/expect-webdriverio/pull/1119). I am not sure of the changes, but it fixed the issue with `test:typings` . Can you please validate those changes and review that PR?

**Comment by sriteja777** [comment]
Thanks @christian-bromann . will do!

**Review by christian-bromann — APPROVED** [review]
Fantastic work 👏  thanks a lot 🙏

**Comment by sriteja777** [comment]
I was debugging this issue and got to know that this PR is merged. Thanks!!

**Comment by christian-bromann** [comment]
@sriteja777 it seems it is not fully fixed. Note that the release pipeline runs with `NODE_ENV: production` as environment variable. Can you try to set it this way and run `npm run setup-full`?

**Comment by sriteja777** [comment]
Yes I got the error when running with NODE_ENV as production, but the build is working fine when that is not as set. 

What do we differently in production and non-production 🤔?

**Comment by sriteja777** [comment]
Ok I got the issue, I have added the `@types/glob` package as dev dependency. But in the production setting dev dependencies are not installed thereby causing the issue. Will create a PR to move it to normal dependency

**Comment by sriteja777** [comment]
@christian-bromann have created a PR for this. Please review 
https://github.com/webdriverio/webdriverio/pull/10179

**Comment by sriteja777** [comment]
@christian-bromann were you able to release v7 after the fix?

**Pull request merged**